x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/check_for_toast.ts

```diff
@@ -5,25 +5,49 @@
  */
 
 import { i18n } from '@kbn/i18n';
+import { ElementHandle } from 'puppeteer';
 import { HeadlessChromiumDriver as HeadlessBrowser } from '../../../../server/browsers/chromium/driver';
+import { LevelLogger as Logger } from '../../../../server/lib/level_logger';
 import { LayoutInstance } from '../../layouts/layout';
 
 export const checkForToastMessage = async (
   browser: HeadlessBrowser,
-  layout: LayoutInstance
-): Promise<void> => {
-  await browser.waitForSelector(layout.selectors.toastHeader, { silent: true });
-  const toastHeaderText = await browser.evaluate({
-    fn: selector => {
-      const nodeList = document.querySelectorAll(selector);
-      return nodeList.item(0).innerText;
-    },
-    args: [layout.selectors.toastHeader],
-  });
-  throw new Error(
-    i18n.translate('xpack.reporting.exportTypes.printablePdf.screenshots.unexpectedErrorMessage', {
-      defaultMessage: 'Encountered an unexpected message on the page: {toastHeaderText}',
-      values: { toastHeaderText },
-    })
-  );
+  layout: LayoutInstance,
+  logger: Logger
+): Promise<ElementHandle<Element>> => {
+  return await browser
+    .waitForSelector(layout.selectors.toastHeader, { silent: true })
+    .then(async () => {
+      // Check for a toast message on the page. If there is one, capture the
+      // message and throw an error, to fail the screenshot.
+      const toastHeaderText: string = await browser.evaluate({
+        fn: selector => {
+          const nodeList = document.querySelectorAll(selector);
+          return nodeList.item(0).innerText;
+        },
+        args: [layout.selectors.toastHeader],
+      });
+
+      // Log an error to track the event in kibana server logs
+      logger.error(
```

Comment:
This sneaks in a new logging message. Previously, having the code only throw the error would make the job fail with that reason, but it would be hard to see what's going on just by looking at the Kibana logs